assets/js/base/hooks/checkout/use-checkout-address.js

```diff
@@ -53,19 +53,31 @@ export const useCheckoutAddress = () => {
 	);
 
 	// Pushes to global state when changes are made locally.
+	// setBillingData and setShippingAddress are memoized from the contexts.
 	useEffect( () => {
 		if ( ! isEqual( shippingFields, shippingAddress ) ) {
 			setShippingAddress( shippingFields );
 		}
 
-		const newBillingData = shippingAsBilling
-			? shippingFields
-			: billingFields;
+		// Uses shipping or billing fields depending on shippingAsBilling checkbox, but ensures
+		// billing only fields are also included.
+		const newBillingData = {
+			...( shippingAsBilling ? shippingFields : billingFields ),
+			email: billingFields.email,
+			phone: billingFields.phone,
+		};
 
 		if ( ! isEqual( newBillingData, billingData ) ) {
 			setBillingData( newBillingData );
 		}
-	}, [ shippingFields, billingFields, shippingAsBilling ] );
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I needed to not include setBillingData and setShippingAddress here. Doing so prevents input.

Reply:
Hmm that smells 🐟  to me. If `setBillingData` and `setShippingAddress` are memoized, then adding them as dependencies should have no effect. If it does, then there's a bug somewhere else.

The good news is I added them back in as dependencies and it tested just fine. When you say "Doing so prevents input", do you mean you are unable to change the value of any inputs? Because if so, I had no problems with that.

Reply:
Let me test again and get back to you. When I added all suggested dependencies earlier (and when I first worked on this) it preventing text input on billing fields like email.

Reply:
I had that issue yesterday too when I was experimenting with this and added all the deps prompted by linter - I couldn't type in any checkout fields. This was in a branch prior to #2467, so I'm guessing there were other fixes upstream that fixed the underlying issue 🎉